var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -79,7 +79,7 @@ class Mpich(AutotoolsPackage):
     # fix MPI_Barrier segmentation fault
     # see https://lists.mpich.org/pipermail/discuss/2016-May/004764.html
     # and https://lists.mpich.org/pipermail/discuss/2016-June/004768.html
-    patch('mpich32_clang.patch', when='@3.2%clang')
+    patch('mpich32_clang.patch', when='@3.2.0%clang')
```

Comment:
There is no version 3.2.0.

@tgamblin I think this is a bug in the concretizer. `@3.2` should only apply to version 3.2, not to every version 3.2.X.

At the same time, I'm really confused, because I've always wanted to say `depends_on('python@2.7')`, but that's never worked because it would install exactly version 2.7.

So now I'm really confused how the concretizer works with regards to MAJOR.MINOR version numbers.

Reply:
My aim was to have 3.2 work and not have 3.2.1 try to patch and bomb.  Empirically this worked on my setup.

Reply:
This change does prevent 3.2.1 from being patched, but it also prevents 3.2 from being patched.

Reply:
It would, but then we would need to override `url_for_version` to get things to work. The underlying bug still remains.